### ISSUE TITLE
Regression in matconvnet

### ISSUE THREAD

**Issue by vamsikrishnagedela** [vlfeat/matconvnet#315]
Hello. I want to use convolutional neural network for regression for facial keypoint detection. I have defined the error function last layer of my network as 'pdist' instead of 'softmax'. What changes should I make in the cnn_train.m file?

**Comment by arunmallya** [comment]
You will actually have to modify the vl_simplenn.m file (if you're using the simple interface) as follows:

Line 220
```matlab
case 'loss'
      res(i+1).x = vl_nnloss(res(i).x, l.class, [], 'loss', l.loss) ;
```

and Line 358
```matlab
case 'loss'
        res(i).dzdx = vl_nnloss(res(i).x, l.class, res(i+1).dzdx, 'loss', l.loss) ;
```

where l.loss should be 'pdist' in the loss structure. 
``` matlab
net.layers{end+1} = struct('type', 'loss',...
                    'loss', 'pdist');
```

Unfortunately, the current code doesn't have the above which allows you to specify the loss function to use. The vl_nnloss() however has the ability to use different loss functions.

**Comment by bazilas** [comment]
Hi,

Please have a look in: #15 or #115 

V
